tests/core/snapd-maintenance-msg/task.yaml

```diff
@@ -0,0 +1,54 @@
+summary: Verify that the maintenance message is included in the daemon API responses when available
+
+# kinda slow test, probably enough to keep on one system
+systems: [ubuntu-core-20-64]
+
+prepare: |
+    # devmode as the snap does not have snapd-control
+    snap install test-snapd-curl --devmode --edge
+    snap install jq
+
+restore: |
+    snap remove test-snapd-curl jq
+
+execute: |
+    if [ "$SPREAD_REBOOT" = 0 ]; then
+        echo "SNAPD_SHUTDOWN_DELAY=1" >> /etc/environment
```

Comment:
@sergiocazzolato does our test framework undo this automatically? or do we need to undo this in somehow?

Reply:
I don't think it does, and it's a good catch - I added cleanup in the restore step